httpreplay/cobweb.py

```diff
@@ -175,3 +180,245 @@ def handle(self, s, ts, protocol, sent, recv):
 
 class SmtpProtocol(Protocol):
     """Interprets the SMTP protocol."""
+
+    def init(self, *args, **kwargs):
+        self.request = SmtpRequest()
+        self.reply = SmtpReply()
+
+        # Current smtp command from client
+        self.command = None
+
+        # Last smtp server response code
+        self.rescode = 0
+
+        # Last smtp server response message
+        self.message = ""
+
+        # Used in handler() to determine if the TCP stream
+        # was closed or is finished
+        self.stream = None
+
+        # Contains the functions to be called when this command is the first
+        # string in a request message
+        self._commands = {
+            "ehlo": self.handle_hostname,
+            "helo": self.handle_hostname,
+            "mail": self.handle_mail,
+            "rcpt": self.handle_rcpt,
+            "auth": self.handle_auth,
+            "etrn": None, "turn": None, "atrn": None, "size": None,
+            "etrn": None, "pipelining": None, "chunking": None, "data":None,
+            "dsn": None, "rset": None, "vrfy": None, "help": None,
+            "quit": None, "noop": None, "expn": None, "binarymime": None,
+            "relay": None, "size": None, "starttls": None, "checkpoint": None,
+            "enhancedstatuscodes": None, "8bitmime": None, "send": None
```

Comment:
If you get rid of all the `None` entries here and.. [go to part 2 of this comment]